site_scons/site_tools/roc/config.py

```diff
@@ -290,6 +295,67 @@ def FindConfigGuess(context):
     context.Result('not found')
     return False
 
+def FindPkgConfigPath(context):
+    env = context.env
+
+    context.Message("Searching PKG_CONFIG_PATH...")
+
+    if env.HasArg('PKG_CONFIG_PATH'):
+        context.Result(env['PKG_CONFIG_PATH'])
+        return True
+
+    if env.HasArg('PKG_CONFIG'):
+        pkg_config = env['PKG_CONFIG']
+    elif env.Which('pkg-config'):
+        pkg_config = 'pkg-config'
+    else:
+        context.Result('not found')
+        return False
+
+    pkg_config_paths = env.CommandOutput('%s --variable pc_path pkg-config' % quote(pkg_config))
+    try:
+        for path in pkg_config_paths.split(':'):
+            if os.path.isdir(path):
+                env['PKG_CONFIG_PATH'] = path
+                break
+    except:
+        pass
+    finally:
+        if 'PKG_CONFIG_PATH' not in env.Dictionary():
+            # https://linux.die.net/man/1/pkg-config the default is libdir/pkgconfig
+            env['PKG_CONFIG_PATH'] = os.path.join(env['ROC_SYSTEM_LIBDIR'], 'pkgconfig')
+            if not os.path.isdir(env['PKG_CONFIG_PATH']):
+                os.makedirs(env['PKG_CONFIG_PATH'])
```

Comment:
I think FindPkgConfigPath() should not call makedirs because it's task is to *find* the directory. We already have makedirs call in GeneratePkgConfig, so I guess we can just remove this `if`.